modules/Bio/EnsEMBL/RNAProduct.pm

```diff
@@ -0,0 +1,976 @@
+  Arg [-SEQ]          : Manually sets the nucleotide sequence of this
+                        rnaproduct. May be useful if this rnaproduct is not
+                        stored in a database.
+  Arg [-CREATED_DATE] : the date the rnaproduct was created
+  Arg [-MODIFIED_DATE]: the date the rnaproduct was modified
+  Example    : my $rp = Bio::EnsEMBL::RNAProduct->new(
+                 -SEQ_START => 36,
+                 -SEQ_END   => 58
+               );
+  Description: Constructor.  Creates a new RNAProduct object
+  Returntype : Bio::EnsEMBL::RNAProduct
+  Exceptions : none
+  Caller     : general
+  Status     : In Development
+
+=cut
+
+sub new { ## no critic (Subroutines::RequireArgUnpacking)
+  my $caller = shift;
+
+  my $class = ref($caller) || $caller;
+
+  my $type_code = Bio::EnsEMBL::Utils::RNAProductTypeMapper::mapper()
+    ->class_to_type_code($class);
+
+  my ($seq_start, $seq_end, $start_exon, $end_exon, $stable_id, $version, $dbID,
+      $adaptor, $seq, $created_date, $modified_date ) =
+	rearrange(["SEQ_START", "SEQ_END", "START_EXON", "END_EXON",
+                   "STABLE_ID", "VERSION", "DBID", "ADAPTOR", "SEQ",
+                   "CREATED_DATE", "MODIFIED_DATE"], @_);
+
+  # For consistency between stable_id() and stable_id_version()
+  $stable_id //= '';
+
+  # Default version
+  $version //= 1;
+
+  my $self = bless {
+    'start'      => $seq_start,
+    'end'        => $seq_end,
+    'start_exon' => $start_exon,
+    'end_exon'   => $end_exon,
+    'stable_id'  => $stable_id,
+    'version'    => $version,
+    'dbID'       => $dbID,
+    'seq'        => $seq,
+    'created_date' => $created_date,
+    'modified_date' => $modified_date,
+    'type_code'  => $type_code,
+  }, $class;
+
+  $self->adaptor($adaptor);
+
+  return $self;
+}
+
+
+=head2 add_Attributes
+
+  Arg [1..N] : Bio::EnsEMBL::Attribute $attribute
+               Attributes to add.
+  Example    : $rnaproduct->add_Attributes($selenocysteine_attribute);
+  Description: Adds an Attribute to the RNAProduct.
+               If you add an attribute before you retrieve any from database,
+               lazy load will be disabled.
+  Returntype : none
+  Exceptions : throw on incorrect arguments
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub add_Attributes {
+  my ($self, @attribs) = @_;
+
+  if (! exists $self->{'attributes'}) {
+    $self->{'attributes'} = [];
+  }
+
+  for my $attrib (@attribs) {
+    if (! $attrib->isa("Bio::EnsEMBL::Attribute")) {
+      throw("Argument to add_Attribute must be a Bio::EnsEMBL::Attribute");
+    }
+    push (@{$self->{'attributes'}}, $attrib);
+    # FIXME: why is this here???
+    $self->{seq}=undef;
+  }
+
+  return;
+}
+
+
+=head2 add_DBEntry
+
+  Arg [1]    : Bio::EnsEMBL::DBEntry $dbe
+               The dbEntry to be added
+  Example    : $rnaproduct->add_DBEntry($xref);
+  Description: Associates a DBEntry with this rnaproduct. Note that adding
```

Comment:
rnaproduct => RNA product
or should it be $rnaproduct or RNAProduct as used in previous pods?